providers/azure/connection/azureinstancesnapshot/snapshot.go

```diff
@@ -214,12 +214,14 @@ func (sc *snapshotCreator) attachDisk(targetInstance instanceInfo, diskName, dis
 		return err
 	}
 	attachOpt := compute.DiskCreateOptionTypesAttach
+	deleteOpt := compute.DiskDeleteOptionTypesDelete
 	// the Azure API requires all disks to be specified, even the already attached ones.
 	// we simply attach the new disk to the end of the already present list of data disks
 	disks := targetInstance.vm.Properties.StorageProfile.DataDisks
 	disks = append(disks, &compute.DataDisk{
```

Comment:
Just to be sure without reading the complete code:
This only affects disks we create for snapshots? Not, that we set the delete option for a disk the user created.

Reply:
Yeah, we only set that to disks that we attach. Disks that are already there are unaffected